PQAnalysis/io/traj_file/trajectory_reader.py

```diff
@@ -410,25 +420,55 @@ def window_generator(
             # yield the next window
             yield window.copy()
 
-    def calculate_number_of_frames(self) -> int:
+    def calculate_frame_size(self, filename: str = None) -> int:
         """
-        Calculates the number of frames in the trajectory file.
+        Calculates the size of the frame in the trajectory file.
 
         Returns
         -------
         int
-            The number of frames in the trajectory file.
+            The size of the frame in the trajectory file.
 
         Raises
         ------
         TrajectoryReaderError
             If the number of atoms in the first line of the file is invalid.
+        """
+
+        with open(filename, "r", encoding="utf-8") as f:
+            try:
+                n_atoms = int(f.readline().split()[0])
+            except (ValueError, IndexError):
+                self.logger.error(
+                    (
+                        "Invalid number of atoms in the first line "
+                        f"of file {filename}."
+                    ),
+                    exception=TrajectoryReaderError,
+                )
+
+        return n_atoms + 2
+
+    def calculate_number_of_frames(self) -> List[int]:
```

Comment:
```suggestion
    def calculate_number_of_frames_per_file(self) -> List[int]:
```

I would add here really explicitly that it is a per file method, so that the user does not expect to get all of the frames back from the method!

Reply:
I agree